rdr_service/api/resource_api.py

```diff
@@ -0,0 +1,290 @@
+            try:
+                _data = json.loads(request.data.decode('utf-8'))
+                if not isinstance(_data, list):
+                    raise BadRequest('Request body must be list of resource ids.')
+                # Convert request body to list of resource ids.
+                self.batch_ids = list()
+                for _id in _data:
+                    if isinstance(_id, str) or isinstance(_id, int):
+                        self.batch_ids.append(_id)
+                if not self.batch_ids:
+                    raise BadRequest('Invalid batch request data.')
+            except json.JSONDecodeError:
+                raise BadRequest('Invalid batch request data.')
+
+        self.uri_args = request.values
+
+        mod = getattr(RESOURCE_SCHEMAS, self.schema_name)
+        self.schema = mod()
+
+
+class ResourceRequestApi(Resource):
+    """
+    Resource Request API
+    Resource Request Syntax:
+        METHOD /[prefix]/[resource]/{[suffix]|/?[argument{:modifiers}={prefix}value,...]}
+    """
+    dao = ResourceDataDao()
+
+    @auth_required(RESOURCE)
+    def get(self, path):  # pylint: disable=unused-argument
+        """
+        Handle GET requests
+        :param path: URI request path
+        :return:
+        """
+        resource = RequestResource()
+
+        if resource.suffix == '_schema':
+            resp = self._get_schema(resource)
+        elif resource.suffix == '_meta':
+            resp = self._get_meta(resource)
+        elif resource.suffix == '_search':
+            raise BadRequest('Not Implemented Yet')
+        elif resource.suffix == '_batch':
+            resp = self._get_batch(resource)
+        else:
+            resp = self._get_resource(resource)
+
+        # TODO: Handle request argument modifiers, IE: _sort, _id.
+        # TODO: Handle awardee filtering (probably very last todo item).
+
+        return resp
+
+    def _get_resource(self, resource):
+        """
+        Return the requested resource.
+        :param resource: RequestResource object
+        :return: Meta data json.
+        """
+        if not resource.pk_id and not resource.pk_alt_id:
+            raise BadRequest('Resource request without Resource id.')
+
+        with self.dao.session() as session:
+
+            # Return the meta data for the requested Resource Data record.
+            for _id in [resource.pk_id, resource.pk_alt_id]:
+                query = session.query(ResourceData.id, ResourceData.created, ResourceData.modified,
+                                      ResourceData.resource)
+                if isinstance(_id, int):
+                    query = query.filter(ResourceData.resourcePKID == _id)
+                elif isinstance(_id, str):
+                    query = query.filter(ResourceData.resourcePKAltID == _id)
+                else:
+                    raise ValueError(f'Invalid resource id: {_id}.')
+                # sql = self.dao.query_to_text(query)
```

Comment:
Was this left over from testing?

Reply:
I like to leave those commented out lines there for testing in the future.